geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/QueryPlanners.scala

```diff
@@ -304,11 +317,11 @@ trait GeoHashPlanner extends Logging {
     case SpatialFilter(geom)                => polyToPlan(geom, offset, bits)
     case SpatialDateFilter(geom, _)         => polyToPlan(geom, offset, bits)
     case SpatialDateRangeFilter(geom, _, _) => polyToPlan(geom, offset, bits)
-    case DateRangeFilter(_, _)              => KeyAccept
-    case AcceptEverythingFilter             => KeyAccept
+    case DateRangeFilter(_, _)              => KeyAccept(bits)
+    case AcceptEverythingFilter             => KeyAccept(bits)
```

Comment:
It's not really for table sharing - but in order to join keys in tiers, keyAccept needs to know how many chars it contains. I'm not sure it was really affecting us in the default case, but in the time first index work I did I noticed it was broken, so I brought over the fix.